### PR TITLE
chore: Add rust-toolchain file and bump to 1.37

### PR DESCRIPTION
**Pull request by LucioFranco**
Signed-off-by: Lucio Franco <luciofranco14@gmail.com>

### CODE REVIEW

**Comment by LucioFranco** [comment]
This PR also applies cargo fix --edition-idioms

File: lib/tracing-limit/benches/limit.rs [hunk 1]
```diff
@@ -102,18 +102,18 @@ impl<'a> field::Visit for Visitor<'a> {
 }
 
 impl tracing::Subscriber for VisitingSubscriber {
-    fn new_span(&self, span: &span::Attributes) -> Id {
+    fn new_span(&self, span: &span::Attributes<'_>) -> Id {
```
Comment (lukesteensen) [inline lib/tracing-limit/benches/limit.rs]: What's the deal with these?

Comment (LucioFranco) [reply lib/tracing-limit/benches/limit.rs]: The elided lifetimes are part of 2018 edition idioms.

Comment (lukesteensen) [reply lib/tracing-limit/benches/limit.rs]: [Found it](https://doc.rust-lang.org/edition-guide/rust-2018/ownership-and-lifetimes/the-anonymous-lifetime.html#more-details), interesting

**Pull request merged**